akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/EventSourcedBehavior.scala

```diff
@@ -141,26 +120,40 @@ abstract class EventSourcedBehavior[Command, Event, State >: Null] private[akka]
    */
   def snapshotSelectionCriteria: SnapshotSelectionCriteria = SnapshotSelectionCriteria.latest
 
+  /**
+   * Initiates a snapshot if the given function returns true.
+   * When persisting multiple events at once the snapshot is triggered after all the events have
+   * been persisted.
+   *
+   * receives the State, Event and the sequenceNr used for the Event
+   *
+   * Snapshots triggered by `snapshotWhen` will not trigger deletes of old snapshots and events if
+   * [[EventSourcedBehavior.retentionCriteria]] with [[RetentionCriteria.snapshotEvery]] is used together with
+   * `shouldSnapshot`. Such deletes are only triggered by snapshots matching the `numberOfEvents` in the
+   * [[RetentionCriteria]].
+   *
+   * @return `true` if snapshot should be saved for the given event
```

Comment:
? May sound like we snapshot for one event
```suggestion
   * @return `true` if snapshot should be saved at the given event
```